### ISSUE TITLE
API: creating an experiment and adding trees to an experiment aren't atomic

### ISSUE THREAD

**Issue by brachbach** [oughtinc/mosaic#723]
If there's a problem with one of the trees that's supposed to be created (in practice, usually an email address for a malicious expert that's not yet in the app), the trees will be created up until that tree, then the tree creation will error, leaving all of the trees created so far around

It would be good if either all of the trees were created, or none

(atomic as in: https://en.wikipedia.org/wiki/ACID#Atomicity)

**Comment by brachbach** [comment]
BTW I thought I had created this issue a while ago, so may be a duplicate

**Comment by zjmiller** [comment]
Yeah good point would make sense to wrap in a transaction.

**Comment by andrewschreiber** [comment]
Agreed as well. @zjmiller do we have any examples of transactions with atomic guarantees in Mosaic?

**Comment by zjmiller** [comment]
No but in this case with everything limited to a single function body is should be fairly straightforward: https://sequelize.org/master/manual/transactions.html

**Comment by andrewschreiber** [comment]
For changing expert designations, https://github.com/oughtinc/mosaic/pull/701 now uses atomic transactions. I will save atomic transactions on `bulkCreateTrees` for a separate PR so 701 doesn't get too large.
